chrome/content/zotero/xpcom/pdfWorker/manager.js

```diff
@@ -248,28 +261,50 @@ class PDFWorker {
 			buf = new Uint8Array(buf).buffer;
 
 			try {
-				var { imported, deleted } = await this._query('import', {
-					buf, existingAnnotations, password
+				var { imported, deleted, buf: modifiedBuf } = await this._query('import', {
+					buf, existingAnnotations, password, transfer
 				}, [buf]);
 			}
 			catch (e) {
 				let error = new Error(`Worker 'import' failed: ${JSON.stringify({
 					existingAnnotations,
 					error: e.message
 				})}`);
+				try {
+					error.name = JSON.parse(e.message).name;
+				}
+				catch (e) {
+					Zotero.logError(e);
+				}
 				Zotero.logError(error);
 				throw error;
 			}
-
-			for (let annotation of imported) {
-				annotation.key = Zotero.DataObjectUtilities.generateKey();
-				annotation.isExternal = true;
-				await Zotero.Annotations.saveFromJSON(attachment, annotation);
-			}
-
+			
+			let ids = [];
 			for (let key of deleted) {
 				let annotation = Zotero.Items.getByLibraryAndKey(attachment.libraryID, key);
-				await annotation.eraseTx();
+				if (annotation) {
+					ids.push(annotation.id);
+				}
+			}
+			if (ids.length) {
+				await Zotero.Items.erase(ids);
+			}
+			
+			await Zotero.DB.executeTransaction(async function () {
+				for (let annotation of imported) {
+					annotation.key = Zotero.DataObjectUtilities.generateKey();
+					annotation.isExternal = !(transfer && annotation.transferable);
+					annotation.tags = annotation.tags.map(x => ({ name: x }));
+					await Zotero.Annotations.saveFromJSON(attachment, annotation);
+				}
+			});
```

Comment:
Needs investigation why this transaction running for a few minutes results to issues like this:
```
zotero(1)(+0000002): TypeError: item.isRegularItem is not a function

    TypeError: item.isRegularItem is not a function
        _updateItem@chrome://zotero/content/xpcom/retractions.js:839:10
        notify@chrome://zotero/content/xpcom/retractions.js:393:9
        Zotero.Notifier</this.trigger<@chrome://zotero/content/xpcom/notifier.js:140:38
    From previous event:
        import/<@chrome://zotero/content/xpcom/pdfWorker/manager.js:294:13
```